src/server/races/default-race.js

```diff
@@ -0,0 +1,291 @@
+
+		this.round = new Round(levelId);
+
+		// Contains a set of functions that returns a valid value if the passed attribute string can be parsed correctly
+		// Attributes/functions can be replaced/removed/added as needed
+		this.marbleAttributeList = {
+			color: {
+				userValue: (userId, attribute) => {
+					const colorRegEx = /#(?:[0-9a-fA-F]{3}){1,2}$/g;
+					let match = attribute.match(colorRegEx);
+					return match === null ? undefined : match[0];
+				},
+				defaultValue: () => {
+					return utility.randomHexColor();
+				}
+			},
+			size: {
+				defaultValue: () => {
+					// 2% chance for a entry's marbles to be of a random size between .3 and .6
+					return Math.random() > .98 ? (.3 + Math.random() * .3) : 0.2;
+				}
+			},
+			skinId: {
+				userValue: (userId, attribute) => {
+					let skinId = skins.idList[attribute];
+					if (typeof skinId === "undefined")
+						return undefined;
+
+					// Check if this user has permission to use this skin
+					if(	config.discord.enabled
+						&& skins.skinList[skinId].premium
+						&& !permissions.memberHasPermission(userId, "PREMIUM_SKINS")) {
+						return undefined;
+					}
+					return skinId;
+				},
+				defaultValue: () => {
+					return "default";
+				}
+			}
+		};
+	}
+
+	//
+	// State-related functions
+	//
+
+	onStateEnter() {
+	}
+
+	// Returns a promise that resolves once preparation has finished, or null if this state can be skipped
+	onStatePreparing() {
+		return null;
+	}
+
+	onStateStarting() {
+		// Default point contribution rules
+		// Other race types may want to override this with their own
+		this.pointContributors = this.raceEntries.filter(entry => entry.contributesPoints).length;
+	}
+
+	onStateStarted() {
+		this.round.start = Date.now();
+		physicsWorld.openGates();
+	}
+
+	onStateFinished() {
+		this.finalizeRound();
+
+		physicsWorld.clearMarbles();
+		physicsWorld.closeGates();
+	}
+
+	//
+	// Non-state functions
+	//
+
+	// Returns true if the marble/entry limit has been reached
+	addRaceEntry(id, name, attributeString = "") {
+		// Don't add new player entries if the limit has been reached, or players that already entered
+		// Exception: Added bots with a undefined id (only spawnable by devs/operators)
+		if(typeof id === "string" && this.isRaceFull()
+		|| (typeof id === "string" && typeof this.raceEntries.find(playerEntry => id === playerEntry.id) !== "undefined") ) {
+			return this.isRaceFull();
+		}
+
+		let marbleAttributes = this.parseMarbleAttributes(id, attributeString);
+		let entry = new RaceEntry(id, name, marbleAttributes);
+		this.raceEntries.push(entry);
+
+		this.onNewRaceEntry(entry);
+
+		return this.isRaceFull();
+	}
+
+	// Returns a marbleAttributes object for the given user id and attributes string
+	parseMarbleAttributes(id, attributeString) {
+		let marbleAttributes = {};
+		let messageSections = attributeString.split(" ").filter( (attr) => {return attr != "";}); // Split, filter out additional whitespaces
```

Comment:
You can clean up the filter callback a bit:
`.filter(attr => attr != "")`

I'm also wondering about the non-strict check, can't we use `!==` in this situation?